framework/configuration/rest-resource.rst

```diff
@@ -0,0 +1,203 @@
+It defines a method called ``getEvents()`` and a second method ``addEvent()``. The first method is mapped to
+`GET` requests, the second one to `POST` requests. Both methods return JSON results.
+
+.. hint::
+
+   ${BASE_PATH} is ``/nice/rest``. So the full path of the following resource is ``/nice2/rest/events/{city}``.
+
+.. code-block:: Java
+
+   @Path("/events/{city}")
+   public interface EventsResource extends RestResource {
+
+       @GET
+       @Produces(MediaType.APPLICATION_JSON)
+       @Operation(
+           summary = "Load events",
+           description = "Load events which take place in a certain city",
+           tags = "events"
+       )
+       CollectionBean getEvents(
+           @PathParam("city") @Parameter(description = "name of the city") String city,
+           @QueryParam("sort") @Parameter(description = "comma separated string of fields to sort by") String sort
+       );
+
+       @POST
+       @Consumes(MediaType.APPLICATION_JSON)
+       @Produces(MediaType.APPLICATION_JSON)
+       @Operation(
+           summary = "Create event",
+           description = "Create a new event",
+           tags = "events"
+       )
+       Response addEvent(EventBean event)
+   }
+
+There is an extensive set of **JAX-RS** annotations which can be used to define the behavior of a resource:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Annotation
+     - Description
+   * - Path
+     - Identifies the URI path. Can be specified on a class or a method.
+   * - PathParam
+     - Represents the parameter of the URI path.
+   * - GET
+     - Specifies the method that responds to GET requests.
+   * - POST
+     - Specifies the method that responds to POST requests.
+   * - PUT
+     - Specifies the method that responds to PUT requests.
+   * - ch.tocco.nice2.rest.core.spi.PATCH
+     - Specifies the method that responds to PATCH requests (note that this annotation is not part of the
+       ``javax.ws.rs`` package).
+   * - HEAD
+     - Specifies the method that responds to HEAD requests.
+   * - DELETE
+     - Specifies the method that responds to DELETE requests.
+   * - OPTIONS
+     - Specifies the method that responds to OPTIONS requests.
+   * - FormParam
+     - Represents the parameter of the form.
+   * - QueryParam
+     - Represents the parameter of the query string of an URL.
+   * - HeaderParam
+     - Represents the parameter of the header.
+   * - CookieParam
+     - Represents the parameter of the cookie.
+   * - Produces
+     - Defines the media type for the response such as XML, PLAIN, JSON etc.
+   * - Consumes
+     - Defines the media type that the method of a resource class can consume.
+
+Swagger documentation
+^^^^^^^^^^^^^^^^^^^^^
+
+There is a Swagger documentation available on ``/nice2/swagger``. Use the annotations ``@Operation`` and ``@Parameter``
+to describe the resource in this documentation.
```

Comment:
Maybe additionally a link to https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations ?